src/components/button/button.ts

```diff
@@ -1,68 +1,79 @@
-import { LitElement, html } from 'lit';
+/**
+ * Copyright Kyndryl, Inc. 2023
+ */
+
+import { html, LitElement } from 'lit';
 import { customElement, property } from 'lit/decorators.js';
-import ButtonScss from './button.scss';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { classMap } from 'lit-html/directives/class-map.js';
+
+import { BUTTON_SIZES, BUTTON_TYPES, BUTTON_ICON_POSITION } from './defs';
+
+import stylesheet from './button.scss';
 
 /**
- * Primary UI component for user interaction.
- * @fires on-click - Captures the click event and emits the original event details.
- * @csspart button - Styles to override the button
+ * Kyndryl-branded L0 Button component.
+ *
+ * @slot - Content slot for the button.
+ * @csspart button - Styles applied to the button.
  */
-@customElement('kyn-button')
+@customElement('kd-button')
 export class Button extends LitElement {
-  static override styles = ButtonScss;
+  static override styles = [stylesheet];
+
+  /** Identifier used primarily for testing purposes. */
+  @property({ type: String, attribute: 'data-testid' }) dataTestId =
+    'kd-button-testId';
+
+  /** ARIA label for the button for accessibility. */
+  @property({ type: String }) description = '';
 
-  /** The kind of button to render. */
-  @property({ type: String })
-  kind = 'primary';
+  /** Specifies the visual appearance/type of the button. */
+  @property() type: BUTTON_TYPES = BUTTON_TYPES.PRIMARY;
 
-  /** The button size. */
-  @property({ type: String })
-  size = 'md';
+  /** Indicates if the context pertains to an app. */
+  @property({ type: Boolean }) app = true;
 
-  /** Converts to an &lt;a&gt; tag to create a link. */
-  @property({ type: String })
-  href = '';
```

Comment:
`href` prop is still needed. Buttons that are links must be `<a>` tags.

Reply:
Spoke with Suresh. Link is being work on on a separate feature by Maulik.
Since this is a button, it should only handle actions.
Navigation should be Link.

Reply:
Hmm I'm sure people will try to implement buttons that are also links, it always happens. Be aware that enhancement may need to be added later then.